Zend/zend_vm_def.h

```diff
@@ -2090,11 +2090,24 @@ ZEND_VM_HANDLER(98, ZEND_FETCH_LIST, CONST|TMPVAR|CV, CONST|TMPVAR|CV)
 {
 	USE_OPLINE
 	zend_free_op free_op1, free_op2;
-	zval *container;
+	zval *container, *retval_ptr, retval;
+	int type = opline->extended_value == ZEND_LIST_MAKE_WRITABLE ? BP_VAR_RW : BP_VAR_R;
 
 	SAVE_OPLINE();
-	container = GET_OP1_ZVAL_PTR_UNDEF(BP_VAR_R);
-	zend_fetch_dimension_address_read_LIST(EX_VAR(opline->result.var), container, GET_OP2_ZVAL_PTR_UNDEF(BP_VAR_R));
+	container = GET_OP1_ZVAL_PTR_UNDEF(type);
+
+	zend_fetch_dimension_address_LIST(&retval, container, GET_OP2_ZVAL_PTR_UNDEF(BP_VAR_R), type);
+
+	if (UNEXPECTED(Z_TYPE(retval) == IS_INDIRECT)) {
+		retval_ptr = Z_INDIRECT(retval);
+		if (EXPECTED(!Z_ISREF_P(retval_ptr))) {
+			ZVAL_MAKE_REF(retval_ptr);
+		}
+		GC_REFCOUNT(Z_REF_P(retval_ptr))++;
+		ZVAL_REF(EX_VAR(opline->result.var), Z_REF_P(retval_ptr));
+	} else {
+		ZVAL_COPY_VALUE(EX_VAR(opline->result.var), &retval);
+	}
```

Comment:
I was debating doing that as well.  However, since I didn't want to get too expansive on, what seemed to be a smaller change (w/ indirects earlier) I kept the one.  I will split this into the two respective opcodes.

When I do that, my other concern, was for opcode caching.  I'm not overly familiar with that end of the code, and am not sure the direct impact that would have there.  Would you be able to shed any light on the implications of making FETCH_LIST_R/RW?